File: prime/src/main/java/org/ostelco/prime/storage/entities/Subscriber.java
package org.ostelco.prime.storage.entities;

import org.ostelco.prime.storage.AsMappable;

import java.util.HashMap;
import java.util.Map;

public interface Subscriber extends AsMappable {

    long getNoOfBytesLeft();

    String getMsisdn();

    @Override
    default Map<String, Object> asMap() {
        final Map<String, Object> result = new HashMap<>();
        result.put("noOfBytesLeft", getNoOfBytesLeft());
        result.put("msisdn", getMsisdn());
        return result;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/storage/entities/Subscriber.kt
package org.ostelco.prime.storage.entities

import org.ostelco.prime.storage.AsMappable
import java.util.*

interface Subscriber : AsMappable {

    val noOfBytesLeft: Long

    val msisdn: String?

    override fun asMap(): Map<String, Any?> {
        val result = HashMap<String, Any?>()
        result["noOfBytesLeft"] = noOfBytesLeft
        result["msisdn"] = msisdn
        return result
    }
}
